### ISSUE TITLE
Multiple selection in an entity form

### ISSUE THREAD

**Issue by bryxfr** [symfony/symfony#38896]
Hi everyone,

I come to you because I cannot find my answer among the many results. There must be too many. How can I make a multiple selection in my form without SQL query, I mean to have the selected in the HTML?

```
<select required="required" class="form-control" multiple="multiple">
<option value="1">Anti-Magie</option>
<option value="2" selected>Obscurité</option>
<option value="3">Eau</option>
<option value="4" selected>Spatiale</option>
<option value="5">Foudre</option>
</select>
```

The values have been well selected. However, as you can see below, this form does not do it.

```
->add('specialite', EntityType::class, [
'label'         => 'Choisir la ou les spécialités',
'row_attr'      => ['class' => 'form-group'],
'attr'          => ['class' => 'form-control'],
'class'         => PersonnagesSpecialites::class,
'choice_label'  => 'titre',
'choice_value'  => 'id',
'multiple'      => true,
'data'          => $options['specialite']
])
```

Thank you for your help,
Sincerely, Bryx

**Comment by derrabus** [comment]
Hello!

Thanks for reaching out. This issue tracker however is not meant for support questions. You can find all offered support channels here: https://symfony.com/support

**Comment by stollr** [comment]
I came to this issue trying to find out why the selection of the entities in a ManyToMany relation (inverse side) did not work. Now I found the reason. One has to disable 'by_reference' option in the field. It should look like this:

```php
$builder->add('specialite', EntityType::class, [
    'label'         => 'Choisir la ou les spécialités',
    'class'         => PersonnagesSpecialites::class,
    'multiple'      => true,
    'by_reference' => false, // this line makes the difference
    // ...
]);
```

If you disable `by_reference` the form's `PropertyPathMapper` will call the adder and remover of your entity.

Here's the link to a [stackoverflow answer](https://stackoverflow.com/a/64872743/1119601) that may help, too.
